jetty-ee10/jetty-ee10-annotations/src/main/config/modules/ee10-annotations.mod

```diff
@@ -8,15 +8,11 @@ ee10
 
 [depend]
 ee10-plus
+annotations
```

Comment:
This reintroduces a module name that we deprecated.
Those that use Jetty 10/11 will find that using `annotations` works, but not in the way they think.
Perhaps this should be `core-annotations`? or `common-annotations`?

Reply:
I don't think that is a big concern.   Most of our core modules do not have a `core-` prefix, so I don't think it is necessary.

Reply:
Yes, but that's not what I'm talking about. :-)

The `annotations.mod` existed in Jetty 9/10/11 to provide webapp level annotations support.

Example -> https://github.com/jetty/jetty.project/blob/jetty-11.0.25/jetty-annotations/src/main/config/modules/annotations.mod

We deprecated that name in Jetty 12 (removed it actually, to intentionally cause an error on start.jar. like we do for all of the module name that are now _env_ specific), so we don't want to re-introduce the old `annotations` name again.

Reply:
We want the error from old use of `--module=annotations` (in `start.d/*.ini`) to inform users that they need to pick a specific _env_ version now.

Reply:
I understand what you are saying, but I don't think it is a big enough concern to pervert the naming of a module.
The module was gone from 12.0, so it is OK to re-add in 12.1.x

Reply:
I don't see this as perverting the name of a module.
The current `annotations` mod in this PR is actually an internal module, one that itself doesn't do anything useful, you have to use the `ee10-annotations` to have something be useful.
This module should probably be `core/annotations` or something similar that doesn't expose it to be directly selectable / enabled / show up in `--list-modules`, etc.